addons/web/static/tests/views/list_tests.js

```diff
@@ -3207,6 +3220,42 @@ QUnit.module('Views', {
         list.destroy();
     });
 
+    QUnit.test('do not sort records when clicking on header with nolabel', async function (assert) {
+        assert.expect(6);
+
+        this.data.foo.fields.foo.sortable = true;
+
+        var nbSearchRead = 0;
+        var list = await createView({
+            View: ListView,
+            model: 'foo',
+            data: this.data,
+            arch: '<tree><field name="foo" nolabel="1"/><field name="bar"/></tree>',
+            mockRPC: function (route) {
+                if (route === '/web/dataset/search_read') {
+                    nbSearchRead++;
+                }
+                return this._super.apply(this, arguments);
+            },
+        });
+
+        assert.strictEqual(nbSearchRead, 1, "should have done one search_read");
```

Comment:
i'd click on the other column as a sanity check (to ensure that the remaining asserts are relevant)